gwcs/tests/test_utils.py

```diff
@@ -72,6 +72,30 @@ def test_unknown_ctype():
     assert_allclose(b, expected[1], atol=10**-8)
 
 
+# https://github.com/spacetelescope/gwcs/issues/139
+def test_transform_hdr_dict():
+    """2D test case from Ginga."""
+    header = {
+        'BITPIX': -32, 'BLANK': -32768, 'BUNIT': 'ADU', 'EXTEND': False,
+        'CD1_1': -5.611e-05, 'CD1_2': 0.0, 'CD2_1': 0.0, 'CD2_2': 5.611e-05,
+        'CDELT1': -5.611e-05, 'CDELT2': 5.611e-05,
+        'CRPIX1': 5276.0, 'CRPIX2': 25.0,
+        'CRVAL1': 299.91736667, 'CRVAL2': 22.68769444,
+        'CTYPE1': 'RA---TAN', 'CTYPE2': 'DEC--TAN',
+        'CUNIT1': 'degree', 'CUNIT2': 'degree', 'EQUINOX': 2000.0,
+        'NAXIS': 2, 'NAXIS1': 2272, 'NAXIS2': 4273,
+        'RA': '19:59:40.168', 'RA2000': '19:59:40.168',
+        'DEC': '+22:41:15.70', 'DEC2000': '+22:41:15.70', 'RADECSYS': 'FK5',
+        'SIMPLE': True, 'WCS-ORIG': 'SUBARU Toolkit'}
+    w = gwutils.make_fitswcs_transform(header, dict_to_fits=True)
+    xy_ans = np.array([120, 100])
+    radec_deg_ans = (300.2308791294835, 22.691653517073615)
+
+    # TODO: Check with Nadia
```

Comment:
@nden , I don't understand

* why `w()` result can't agree for default `rtol=1e-7`
* why `w.inverse()` would give me original XY + 1 and not XY

Am I using it wrong?

Reply:
GWCS by default uses 0-based pixel coordinates. CRPIX values are 1-based.  So if you want to compare the inverse transform, it's going to return 0-based values.

Reply:
Though maybe I'm misunderstanding the problem?

Reply:
@jdavies-st is correct. When you are mixing FITS and gwcs you have to take care of 0- vs 1- based coordinates. The gwcs transform is 0-based. However, the values of crpix are 1-based. If you change the CRPIX values to be 0-based, i.e. subtract 1 from them , you will get agreement (you'll get better agreement in the forward direction too). 

BTW, the header has incorrect FITS WCS to start with. It is not allowed to mix CD matrix and CDELT values and in this case the CDELT values are ignored and assumed to be 1.

I don't want to encourage changes and "fixes" to these functions as I think it's time to implement full conversion between the two objects.